test/Datadog.Trace.ClrProfiler.IntegrationTests/AspNetCore/AspNetCoreMvcTestBase.cs

```diff
@@ -87,7 +87,14 @@ public void RunTraceTestOnSelfHosted(string packageVersion)
                         throw new Exception("Unable to verify whether the server is ready to receive requests.");
                     }
 
-                    Thread.Sleep(500);
+                    var ready = SubmitRequest(aspNetCorePort, "/alive-check");
+
+                    Thread.Sleep(300);
+
+                    if (ready)
+                    {
+                        break;
+                    }
                 }
```

Comment:
In general, I prefer to set a timeout on a `WebRequest` and then use that time span as an indicator of whether the request finished. Have you looked at doing that, and then sleeping for that timeout period plus a little extra? It takes more time but I think having a very short (300ms) sleep is not necessarily going to work when there's a lot of CPU activity on the test machine.

Reply:
I loop because it usually fails immediately. I wait 300ms to try again to see if the server has actually started. If it has started, then the request goes through successfully. If you look at the `maxTimesToCheck` variable, it will try this 20 times. A longer interval is definitely reasonable now that I look at this again, it could take more than 6 seconds for sure, as you say.

Reply:
With pushing it up to 500ms for the intervals, that gives 15 seconds for the server to start, which is plenty of time I think. Better than the current 5 seconds.

Reply:
Happy to put a timeout on the request as well, but in my experience the request comes back nearly right away with an actively denied message.

Reply:
Yeah, sure, a higher value would be better. Reading the code, I could have figured out that it was 6 seconds. Maybe make it a little clearer by setting the max time waiting period, e.g., 15 seconds, and determine how long to sleep based on that and # retries. Just would make it easier to scan and know how long it might go on for. A `while(true)` always leaves me having to do the math otherwise.